File: app/src/main/java/com/example/workoutapp/di/showworkout/ShowWorkoutModule.java
package com.example.workoutapp.di.showworkout;

import android.content.Context;

import com.example.workoutapp.addworkout.AddWorkoutPresenter;
import com.example.workoutapp.model.workout.WorkoutLocalDataSource;
import com.example.workoutapp.model.workout.WorkoutLocalDataSourceImpl;
import com.example.workoutapp.model.workout.WorkoutRepository;
import com.example.workoutapp.model.workout.WorkoutRepositoryImpl;
import com.example.workoutapp.showworkout.ShowWorkoutContract;
import com.example.workoutapp.showworkout.ShowWorkoutPresenter;

import dagger.Module;
import dagger.Provides;
import io.reactivex.disposables.CompositeDisposable;

@Module
public class ShowWorkoutModule {
    @Provides
    ShowWorkoutContract.Presenter providesShowWorkoutPresenter(WorkoutRepository workoutRepository, CompositeDisposable compositeDisposable) {
        return new ShowWorkoutPresenter(workoutRepository, compositeDisposable);
    }

    @Provides
    WorkoutRepository providesWorkoutRepository(WorkoutLocalDataSource workoutLocalDataSource) {
        return new WorkoutRepositoryImpl(workoutLocalDataSource);
    }

    @Provides
    WorkoutLocalDataSource providesWorkoutLocalDataSource(Context context) {
        return new WorkoutLocalDataSourceImpl(context);
    }

}

File: app/src/main/java/com/example/workoutapp/di/showworkout/ShowWorkoutModule.kt
package com.example.workoutapp.di.showworkout

import com.example.workoutapp.domain.showworkout.*
import com.example.workoutapp.domain.workout.WorkoutRepository
import dagger.Module
import dagger.Provides

@Module
class ShowWorkoutModule {

    @Provides
    fun providesGetWorkoutsUseCase(workoutsRepository: WorkoutRepository): GetWorkoutUseCase {
        return GetWorkoutUseCaseImpl(workoutsRepository)
    }

    @Provides
    fun providesSoftDeleteWorkoutUseCase(workoutsRepository: WorkoutRepository): SoftDeleteWorkoutUseCase {
        return SoftDeleteWorkoutUseCaseImpl(workoutsRepository)
    }

    @Provides
    fun providesUndoSoftDeleteWorkoutUseCase(workoutsRepository: WorkoutRepository): UndoSoftDeleteWorkoutUseCase {
        return UndoSoftDeleteWorkoutUseCaseImpl(workoutsRepository)
    }

}